nowledge-mem-hermes/provider.py

```
@@ -439,16 +474,52 @@ def _normalize_tool_result(result: Any) -> Any:
+    @staticmethod
+    def _resolve_space(config: Dict[str, Any], kwargs: Dict[str, Any]) -> str | None:
+        configured_space = str(config.get("space") or "").strip()
+        if configured_space:
+            return configured_space
+
+        identity = str(kwargs.get("agent_identity", "default") or "default").strip()
+        identity_map = config.get("space_by_identity")
+        if isinstance(identity_map, str):
+            try:
+                identity_map = json.loads(identity_map)
+            except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)